lib/spack/spack/cmd/create.py

```diff
@@ -184,14 +187,27 @@ class CMakePackageTemplate(PackageTemplate):
     """Provides appropriate overrides for CMake-based packages"""
 
     base_class_name = 'CMakePackage'
-
     body = """\
     def cmake_args(self):
         # FIXME: Add arguments other than
         # FIXME: CMAKE_INSTALL_PREFIX and CMAKE_BUILD_TYPE
         # FIXME: If not needed delete this function
         args = []
         return args"""
+    def __init__(self, name, url, versions):
+        PackageTemplate.__init__(self, name, url, versions)
+        with spack.stage.Stage(url, name) as variant_stage:
+            variant_stage.fetch()
+            variant_stage.expand_archive()
```

Comment:
I thought about doing this at one time, especially for R packages because they contain a file with all of their dependencies and the exact versions they require. Python sometimes has a `requirements.txt` with similar values. Currently, `spack create` doesn't expand any archives. If we decide to do this, we should do it for all tarballs in a more general location.